api2-war/src/main/java/eu/europeana/api2/v2/web/controller/ObjectController.java

```diff
@@ -354,7 +354,7 @@ private Object handleRecordRequest(RecordType recordType, RequestData data, Http
                 model.put("error", "Non-existing record identifier");
                 result = new ModelAndView("rdf", model);
             } else {
-                result = JsonUtils.toJson(new ApiError(data.wskey, "Invalid record identifier: "
+                result = JsonUtils.toJson(new ApiError(data.servletRequest.getHeader("X-Api-Key"), "Invalid record identifier: "
```

Comment:
we are not sure that the API key value is in the header or in the request param here, this will be null if not passed in the header. 

I think as I see we have to pass the API key in quite a lot of places in ObjectController. I would suggest the parameter in RequestData should not be removed and you can extract the wskey value there once and that can be passed everywhere later